cv-data-controller/src/main/java/com/trihydro/cvdatacontroller/controller/ActiveTimController.java

```diff
@@ -47,1550 +48,1090 @@
+        List<ActiveTim> activeTims = new ArrayList<>();
+
+        String selectStatement = "select active_tim.* from active_tim";
+        selectStatement += " left join tim_rsu on active_tim.tim_id = tim_rsu.tim_id";
+        selectStatement += " where active_tim.sat_record_id is null";
+        selectStatement += " and tim_rsu.rsu_id is null";
+
+        try (Connection connection = dbInteractions.getConnectionPool(); Statement statement = connection.createStatement(); ResultSet rs = statement.executeQuery(selectStatement)) {
+            activeTims = getActiveTimFromRS(rs, false);
+        } catch (SQLException e) {
+            log.error("Error getting active tims not sent", e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(activeTims);
+        }
+        return ResponseEntity.ok(activeTims);
+    }
+
+    @RequestMapping(value = "/expired", method = RequestMethod.GET)
+    public ResponseEntity<List<ActiveTim>> GetExpiredActiveTims() {
+        List<ActiveTim> activeTims = new ArrayList<ActiveTim>();
+
+        String selectStatement = "select * from ACTIVE_TIM";
+        selectStatement += " WHERE TIM_END <= (NOW() AT TIME ZONE 'UTC')";
+
+        try (Connection connection = dbInteractions.getConnectionPool(); Statement statement = connection.createStatement(); ResultSet rs = statement.executeQuery(selectStatement)) {
+            activeTims = getActiveTimFromRS(rs, false);
+        } catch (SQLException e) {
+            log.error("Error getting expired active tims", e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(activeTims);
+        }
+
+        return ResponseEntity.ok(activeTims);
+    }
+
+    @RequestMapping(value = "/indices-rsu/{rsuTarget}", method = RequestMethod.GET)
+    public ResponseEntity<List<Integer>> GetActiveTimIndicesByRsu(@PathVariable String rsuTarget) {
+
+        List<Integer> indices = new ArrayList<>();
+
+        String selectStatement = "select tim_rsu.rsu_index from active_tim";
+        selectStatement += " inner join tim on active_tim.tim_id = tim.tim_id";
+        selectStatement += " inner join tim_rsu on tim_rsu.tim_id = tim.tim_id";
+        selectStatement += " inner join rsu on rsu.rsu_id = tim_rsu.rsu_id";
+        selectStatement += " inner join rsu_view on rsu.deviceid = rsu_view.deviceid";
+        selectStatement += " where rsu_view.ipv4_address = '" + rsuTarget + "'";
+
+        try (Connection connection = dbInteractions.getConnectionPool(); Statement statement = connection.createStatement(); ResultSet rs = statement.executeQuery(selectStatement)) {
+            // convert to ActiveTim object
+            while (rs.next()) {
+                indices.add(rs.getInt("RSU_INDEX"));
+            }
+        } catch (SQLException e) {
+            log.error("Error getting active tim indices by rsu", e);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(indices);
+        }

```

Comment:
issue(blocking): the raw query construction opens us up to SQL Injection attacks

suggestion(non-blocking): use parameterized queries to prevent SQL Injection attacks
```suggestion
        selectStatement += " where rsu_view.ipv4_address = ?";

        try (
                Connection connection = dbInteractions.getConnectionPool();
                PreparedStatement preparedStatement = connection.prepareStatement(selectStatement)
        ) {
            // Set the parameter safely
            preparedStatement.setString(1, rsuTarget);

            try (ResultSet rs = preparedStatement.executeQuery()) {
                // Convert to collection of indices
                while (rs.next()) {
                    indices.add(rs.getInt("RSU_INDEX"));
                }
            }
        } catch (SQLException e) {
            log.error("Failed to get active tim indices by rsu", e);
            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(indices);
        }
```

Reply:
closing as it's out of scope and can be easily fixed in a more targeted change set